blog.md

```diff
@@ -0,0 +1,166 @@
+
+## Trivia app
+
+I have a very simple Trivia web application that I would like to host on Kubernetes using the GitOps pattern. It contains a simple React frontend and a nodejs backend that queries existing APIs on the internet for fresh trivia questions. 
+
+![](./images/trivia.png)
+
+Even in a simple application such as this one, there's already two microservices since we have a separate dockerfile for the frontend and backend. This would mean that I need to setup CI pipelines for these individually, and one common CD pipeline should be sufficient to deploy the changes to the cluster. 
+
+I'm following the [5 minutes GitOps pipeline with Bedrock](https://github.com/microsoft/bedrock/blob/master/docs/gitops-quickstart.md) guide which creates a skeleton for all my pipelines and a starter HLD. 
+
+My current setup is outlined below:
+
+![](./images/trivia-ci-diagram.png)
+
+### High level definition
+
+Kubernetes manifest files define the final configuration of the deployment on the cluster. Being extremely error prone, helm charts are a great tool for templating Kubernetes resource definitions. In any application, there are multiple microservices (n) and that leads to (n) helm chart configurations. Bedrock uses the concept of a High Level Definition (HLD) that allows you to define the components of your application at a higher level. 
+
+For example, in my Trivia application I need to include the frontend and backend microservices. But at an even higher level, I want to include Traefik in the cluster. I'm looking at having my high level definition at the root folder look something like below: 
+
+```
+name: default-component
+type: component
+subcomponents:
+- name: hello-world-full-stack
+  type: component
+  method: local
+  path: ./hello-world-full-stack
+- name: traefik2
+  source: https://github.com/microsoft/fabrikate-definitions.git
+  method: git
+  path: definitions/traefik2
+```
+
+And the definition at the nested level to be: 
+
+```
+name: hello-world-full-stack
+type: component
+subcomponents:
+- name: backend
+  type: component
+  method: local
+  path: ./backend
+- name: frontend
+  type: component
+  method: local
+  path: ./frontend
+```
+
+Who doesn't love recursion!
+
+[Fabrikate](https://github.com/microsoft/fabrikate) is the tool behind HLD that simplifies the GitOps workflow: it takes this high level description of the deployment, a target environment configuration (eg. QA or PROD) and renders the Kubernetes resource manifests for that deployment by utilizing templating tools such as Helm. As you may have already guessed, it runs as part of the CI/CD pipeline that connects the HLD repository to the final manifest configuration repository! 
+
+### Deployment Rings
```

Comment:
It might be good for use to include an image from the PPT that relates to deployment rings

Reply:
Let me know if this looks good: https://github.com/microsoft/spektate/blob/d71210ce7402486f6657010d36ef37c47a79afa6/blog.md#deployment-rings